xwayland/selection/incoming.c

```diff
@@ -395,23 +382,29 @@ void xwm_handle_selection_notify(struct wlr_xwm *xwm,
 		return;
 	}
 
+	struct wlr_xwm_selection_transfer *transfer =
+		xwm_selection_find_incoming_transfer_by_window(selection, event->requestor);
+	if (!transfer) {
+		return;
+	}
+
 	if (event->property == XCB_ATOM_NONE) {
 		wlr_log(WLR_ERROR, "convert selection failed");
-		xwm_selection_transfer_finish(&selection->incoming);
+		xwm_selection_transfer_finish(transfer);
 	} else if (event->target == xwm->atoms[TARGETS]) {
 		// No xwayland surface focused, deny access to clipboard
 		if (xwm->focus_surface == NULL) {
 			wlr_log(WLR_DEBUG, "denying write access to clipboard: "
 				"no xwayland surface focused");
 			// Would leak this transfer otherwise. Should never happen.
-			assert(selection->incoming.wl_client_fd <= 0);
+			assert(transfer->wl_client_fd <= 0);
```

Comment:
Hm, this should be `assert(wl_client_fd < 0)` after "xwayland/selection: introduce `xwm_selection_transfer_init`", right?

Reply:
Good catch, fixed in the other PR.